### ISSUE TITLE
[Discussion] Implementing Thread with Homebridge

### ISSUE THREAD

**Issue by squishycat92** [homebridge/homebridge#3091]
### Current Situation

Currently, accessories can connect and be exposed to HomeKit using a variety of different protocols (API, LAN, Bluetooth, etc). However, Thread is not something that is currently supported in the base image. In addition, despite the Homebridge server acting as a hub in HomeKit, it does not interact with devices using Thread in any way and cannot act as a border router nor assume the role of a Thread "Leader" in the event that another Thread border router fails.

### Proposed Change

I propose adding Thread support to Homebridge. I am not completely sure how plugins would interact with accessories over Thread, as it would require the accessory to also support Thread. However, I do think that Thread support should be baked in to the base image for other plugins to use. In addition, baking Thread directly into Homebridge opens the possibility of using the Homebridge server/container as a border router that can act as "Leader" of a local Thread network. I believe that this can be added using an open-source implementation of Thread such as [OpenThread](https://openthread.io). This can allow the Homebridge server to strengthen existing Thread networks, such as extending the range of the mesh network and providing a fallback border router in the event that other Thread devices go offline.

Overall, I think that adding support will be an excellent step into the future, especially with many existing devices already implementing Thread support. It will also help improve network response times and decrease congestion, especially with users with an extremely large amount of HomeKit devices on their network.

I intend for this issue to be a discussion hub for all things Thread-related with Homebridge, if this implementation is possible. Thank you!

### Additional Context

Additional Resources:
1. [Thread (Wikipedia)](https://en.wikipedia.org/wiki/Thread_(network_protocol))
2. [Thread for Developers](https://www.threadgroup.org/What-is-Thread/Developers)
3. [OpenThread](https://openthread.io)

**Comment by MartinPoupa** [comment]
Hi! How it looks like with implementation of Thread in Homebridge?

**Comment by ebaauw** [comment]
Implementing Thread with Homebridge doesn't make sense to me.  The promise of Matter is that we wouldn't be needing "HomeKit support for the impatient" anymore, as "all" devices would be talking each other over Matter.

I'm not sure how you envision Thread with Homebridge?  You would need to distinguish between two scenarios:
1. Homebridge emulating Thread-enabled accessories towards HomeKit (HAP over Thread);
2. A Homebridge plugin connecting to Thread-enabled, non-HomeKit devices, to expose these to HomeKit.

In both cases, you'd need a Thread-capabable radio device (most commonly a USB-stick) connected to the server running Homebridge.

Ad 1.  This scenario makes little sense to me, as HomeKit connects to Homebridge accessories over WiFi.  I don't think switching to Thread would provide any benefit, apart from maybe a small reduction in power consumption.  This would be at the pain of needing an additional hardware stick and OS drivers and an still needing an Apple Thread border router to connect to Homebridge.

Ad 2. This scenario make more sense to me, but has nothing to do with Homebridge itself.  It's up to a plugin to connect to devices.  From my experience trying to connect to devices over Bluetooth Low Energy from NodeJS, I wouldn't be looking forward to try and implement this for any of my plugins.  I would expect most Thread-capable devices to support Matter anyways, if and when it comes available.

**Comment by squishycat92** [comment]
I agree with you, it would primarily be up to plugins to support the connection to HomeKit over Thread. However, with the introduction of official Matter support at [WWDC22](https://developer.apple.com/videos/play/wwdc2022/101/) earlier this month, it seems that the list of devices that will support Thread but not Matter will be quite small - accessory manufacturers will likely include Matter + Thread rather than Thread only.

There is also a third case - using the Homebridge base container as a Thread node in order to strengthen a Thread network. Although we don't currently know much about Apple's implementation of Thread networks with other Thread accessories (notably eero doesn't play well with Apple's Thread network), it would be advantageous for people with Thread networks (especially small ones) to have the extra hop point that can be provided using the Homebridge server/container.

It also seems like users will need special radios (such as the USB sticks you mentioned), but how this would work isn't very clear since there are barely any USB sticks for Thread support (I only found [this one](http://76.227.212.195/products/cortet-radio/usb-sticks/thread-usb-sticks)). For now, this issue is definitely on the backburner; we're primarily waiting to see how the final implementation of Matter plays out with HomeKit and the compatibility of Apple's Thread networks with those created by third parties.

**Comment by SamTheGeek** [comment]
> notably eero doesn't play well with Apple's Thread network

Thread versions before Thread 1.3 were vendor specific — meaning that there were separate thread networks for each manufacturer of the border routers. For the HK-specific Thread network, only Apple hardware (specifically the HomePod mini and TV 4K 2nd Generation) is capable of acting as a border router. 

Thread 1.3, coming soon™ (before the release of Matter later this year), allows for the interoperability of Thread networks — that is, a single thread network can connect to border routers from different manufacturers (including Eero's, though Thread 1.3 is only coming to the Pro 6 and Pro 6E at this point). What is yet to be seen is whether a HK-specific Thread client can connect to HK _through solely_ a non-Apple border router — it should theoretically work, but nobody knows yet if it will.

In any case, there do appear to be a number of HK-specific Thread devices — though Eve has committed to upgrading their devices to Matter, Belkin has said their Thread-enabled HK devices won't be, and the Matter support will be through a new SKU. Depending on whether a non-Apple border router will support HK-only thread devices, it might be useful to have a border router that doesn't require owning a HomePod or TV. 

That being said, it is unlikely Apple will make it possible to build your own HK+Thread border router, so this may be a moot point.

**Comment by wimleers** [comment]
Related: https://github.com/homebridge/homebridge/issues/3228

**Comment by Harvie** [comment]
Hi guys! It seems that Matter support was merged. Have any of you sucesfully deployed homebridge with openthread border router?
It sounds like it should work, since OTBR "converts" thread communication to matter which is already supported by homebridge...

**Comment by Apollon77** [comment]
ut what does this have to do with each other? Get a VM or same host and install OpenThread BR ... I do not see this in any kind related to homebridge
